packages/lodestar/src/eth1/impl/ethers.ts

```diff
@@ -143,79 +161,81 @@ export class EthersEth1Notifier extends (EventEmitter as { new(): Eth1EventEmitt
    *
    * Returns true if processing was successful
    */
-  public async processBlock(blockNumber: number): Promise<boolean> {
+  public async processDepositEvents(blockNumber: number, blockDepositEvents: IDepositEvent[]): Promise<boolean> {
     if (!this.started) {
       this.logger.verbose("Eth1 notifier must be started to process a block");
       return false;
     }
-    this.processingBlock = true;
-    this.logger.verbose(`Processing eth1 block ${blockNumber}`);
-    if (blockNumber !== this.lastProcessedEth1BlockNumber + 1) {
-      this.logger.verbose(
-        `eth1 block out of order. expected: ${this.lastProcessedEth1BlockNumber + 1} actual: ${blockNumber}`
-      );
-      this.processingBlock = false;
-      return false;
-    }
-    const block = await this.getBlock(blockNumber);
-    if (!block) {
-      this.logger.verbose(`eth1 block ${blockNumber} not found`);
-      this.processingBlock = false;
-      return false;
-    }
-    // get results
-    const depositEvents  = await this.getDepositEvents(blockNumber);
-    if (depositEvents.length) {
-      this.lastProcessedDepositIndex = depositEvents[depositEvents.length - 1].index;
-      this.logger.verbose(
-        `${depositEvents.length} deposits found, latest depositIndex: ${this.lastProcessedDepositIndex}`
-      );
-    }
+    this.logger.verbose(`Processing deposit events of eth1 block ${blockNumber}`);
     // update state
     await Promise.all([
       // op pool depositData
-      this.db.depositData.batchPut(depositEvents.map((depositEvent) => ({
+      this.db.depositData.batchPut(blockDepositEvents.map((depositEvent) => ({
         key: depositEvent.index,
         value: depositEvent,
       }))),
       // deposit data roots
-      this.db.depositDataRoot.batchPut(depositEvents.map((depositEvent) => ({
+      this.db.depositDataRoot.batchPut(blockDepositEvents.map((depositEvent) => ({
         key: depositEvent.index,
         value: this.config.types.DepositData.hashTreeRoot(depositEvent),
       }))),
     ]);
-    const depositTree = await this.db.depositDataRoot.getTreeBacked(this.lastProcessedDepositIndex);
+    const depositCount = blockDepositEvents[blockDepositEvents.length - 1].index + 1;
```

Comment:
We should probably put notice that this is not 100% correct logic since deposit events can be small chunks or have invalid signatures (so they won't produce validator) but it won't matter much if we start processing blocks earlier